setupbase.py

```diff
@@ -231,3 +234,84 @@ def run(self):
             'Please check the git diff to make sure everything went okay.\n'
             'You should also edit README.md and '
             'qtawesome/docs/source/usage.rst to reflect the changes.')
+
+
+class UpdateCodiconCommand(distutils.cmd.Command):
+    """A custom command to make updating Microsoft's Codicons easy!"""
+    description = 'Try to update the Codicon font data in the project.'
+    user_options = []
+
+    CHARMAP_PATH = os.path.join(HERE, 'qtawesome', 'fonts', 'codicon.json')
+    TTF_PATH = os.path.join(HERE, 'qtawesome', 'fonts', 'codicon.ttf')
+    DOWNLOAD_URL_TTF = 'https://raw.githubusercontent.com/microsoft/vscode-codicons/master/dist/codicon.ttf'
+    DOWNLOAD_URL_CSV = 'https://raw.githubusercontent.com/microsoft/vscode-codicons/master/dist/codicon.csv'
+    # At the time of writing this comment, vscode-codicons repo does not use git tags, but you can get the version from package.json:
+    DOWNLOAD_URL_JSON = 'https://raw.githubusercontent.com/microsoft/vscode-codicons/master/package.json'
+
+    def initialize_options(self):
+        """Required by distutils."""
+
+    def finalize_options(self):
+        """Required by distutils."""
+
+    def __print(self, msg):
+        """Shortcut for printing with the distutils logger."""
+        self.announce(msg, level=distutils.log.INFO)
+
+    def run(self):
+        """Run command."""
+
+        # Download .csv to a temporary path:
+        package_json = urlopen(self.DOWNLOAD_URL_JSON)
+        package_info = json.load(package_json)
+        package_version = package_info['version']
+        self.__print('Will download codicons version: %s' % package_version)
+
+        # Download .csv to a temporary path:
+        with tempfile.NamedTemporaryFile(mode='wb+', suffix='.csv', prefix='codicon', delete=False) as tempCSV:
+            self.__print('Downloading: %s' % self.DOWNLOAD_URL_CSV)
+            response = urlopen(self.DOWNLOAD_URL_CSV)
+            shutil.copyfileobj(response, tempCSV)
+
+        # Interpret the codicon.csv file:
+        charmap = {}
+        with open(tempCSV.name, 'r') as tempCSV:
```

Comment:
Checking on Windows, running the update option caused an `UnicodeDecodeError`. Adding the `enconding` kwarg helped here.

```suggestion
        with open(tempCSV.name, 'r', encoding='utf-8') as tempCSV:
```